libafl/src/events/launcher.rs

```diff
@@ -183,9 +184,11 @@ pub struct Launcher<'a, CF, MT, SP> {
     /// Tell the manager to serialize or not the state on restart
     #[builder(default = LlmpShouldSaveState::OnRestart)]
     serialize_state: LlmpShouldSaveState,
+    #[builder(default = PhantomData)]
+    phantom: PhantomData<SHM>,
```

Comment:
Adding new phantom data probably means we're holding it wrong

Reply:
So I think here Launcher should be `SP` (with no trait bounds), drop PhantomData, and only have `SHM` below in the implementation where needed